test/llvm/abi/aarch64/export-vs-c-llvm18.good

```diff
@@ -0,0 +1,206 @@
+define %struct.c_one_double @struct_one_double_return_chapel()
+define %struct.c_one_float @struct_one_float_return_c_____()
+define %struct.c_one_float @struct_one_float_return_chapel()
+define %struct.c_pair_double @struct_pair_double_return_c_____()
+define %struct.c_pair_double @struct_pair_double_return_chapel()
+define %struct.c_pair_float @struct_pair_float_return_c_____()
+define %struct.c_pair_float @struct_pair_float_return_chapel()
+define [2 x i64] @struct_paddingtest_return_c_____()
+define [2 x i64] @struct_paddingtest_return_chapel()
+define [2 x i64] @struct_pair_long_return_c_____()
+define [2 x i64] @struct_pair_long_return_chapel()
+define double @double_return_c_____()
+define double @double_return_chapel()
+define float @float_return_c_____()
+define float @float_return_chapel()
+define i32 @int32_return_c_____()
+define i32 @int32_return_chapel()
+define i32 @struct_one_int_return_c_____()
+define i32 @struct_one_int_return_chapel()
+define i32 @uint32_return_c_____()
+define i32 @uint32_return_chapel()
+define i64 @int64_return_c_____()
+define i64 @int64_return_chapel()
+define i64 @struct_one_long_return_c_____()
+define i64 @struct_one_long_return_chapel()
+define i64 @struct_pair_int_return_c_____()
+define i64 @struct_pair_int_return_chapel()
+define i64 @uint64_return_c_____()
+define i64 @uint64_return_chapel()
+define signext i16 @int16_return_c_____()
+define signext i16 @int16_return_chapel()
+define signext i8 @int8_return_c_____()
+define signext i8 @int8_return_chapel()
+define void @abistructfunc_c_____(i32 noundef, i32 noundef, [2 x i64], i32 noundef, i32 noundef, double noundef, double noundef, double noundef, i32 noundef, i32 noundef, i32 noundef0)
+define void @abistructfunc_chapel(i32, i32, [2 x i64], i32, i32, double, double, double, i32, i32, i320)
+define void @bool_arg_c_____(i1 noundef zeroext)
+define void @bool_arg_chapel(i1 zeroext)
+define void @complexd_arg_c_____([2 x double] noundef)
+define void @complexd_arg_chapel([2 x double])
+define void @complexf_arg_c_____([2 x float] noundef)
+define void @complexf_arg_chapel([2 x float])
+define void @double_arg_c_____(double noundef)
+define void @double_arg_chapel(double)
+define void @float_arg_c_____(float noundef)
+define void @float_arg_chapel(float)
+define void @int16_arg_c_____(i16 noundef signext)
+define void @int16_arg_chapel(i16 signext)
+define void @int32_arg_c_____(i32 noundef)
+define void @int32_arg_chapel(i32)
+define void @int64_arg_c_____(i64 noundef)
+define void @int64_arg_chapel(i64)
+define void @int8_arg_c_____(i8 noundef signext)
+define void @int8_arg_chapel(i8 signext)
+define void @struct_aligned16test_arg_c_____(ptr noundef)
+define void @struct_aligned16test_arg_chapel(ptr)
+define void @struct_aligned16test_return_c_____(ptr dead_on_unwind noalias writable sret(%struct.aligned16test) align 16)
+define void @struct_aligned16test_return_chapel(ptr noalias sret(%struct.aligned16test) align 16)
```

Comment:
I would expect we need to match, so `writable` might be needed (I don't know what it does / why it is there). This occurs again in a few of the following functions. It does not seem to be present in the current aarch64 .good file for this test. So I think you should look into it.

Arguably `dead_on_unwind` is in the same boat.

Reply:
I have added `writeable` and `dead_on_unwind` whenever `sret` is added for llvm 18, to match C.